src/core/modules/Planet.cpp

```diff
@@ -682,43 +683,53 @@ QString Planet::getInfoString(const StelCore* core, const InfoStringGroup& flags
 				StelUtils::rectToSphe(&ra_equ,&dec_equ, getEquinoxEquatorialPos(core1));
-				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaMoon, &beta);
-				StelUtils::rectToSphe(&ra_equ,&dec_equ, ssystem->getSun()->getEquinoxEquatorialPos(core1));
-				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaSun, &beta);
-				double deltaLong = (lambdaMoon - lambdaSun)*M_180_PI;
+				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaMoon, &betaMoon);
+				StelUtils::rectToSphe(&raSun,&deSun, ssystem->getSun()->getEquinoxEquatorialPos(core1));
+				StelUtils::equToEcl(raSun, deSun, eclJDE, &lambdaSun, &betaSun);
 				core1->setUseTopocentricCoordinates(state);
 				core1->update(0); // enforce update cache for avoid odd selection of Moon details!
+				double deltaLong = (lambdaMoon-lambdaSun)*M_180_PI;
 				if (deltaLong<0.) deltaLong += 360.;
-				int deltaLongI = qRound(deltaLong);
-				if (deltaLongI==45)
+				const int deltaLongI = static_cast<int>(std::round(deltaLong));
+				if (deltaLongI==0 || deltaLongI==360)
+					moonPhase = qc_("New Moon", "Moon phase");
+				else if (deltaLongI<90)
 					moonPhase = qc_("Waxing Crescent", "Moon phase");
-				if (deltaLongI==90)
+				else if (deltaLongI==90)
 					moonPhase = qc_("First Quarter", "Moon phase");
-				if (deltaLongI==135)
+				else if (deltaLongI<180)
 					moonPhase = qc_("Waxing Gibbous", "Moon phase");
-				if (deltaLongI==180)
+				else if (deltaLongI==180)
 					moonPhase = qc_("Full Moon", "Moon phase");
-				if (deltaLongI==225)
+				else if (deltaLongI<270)
 					moonPhase = qc_("Waning Gibbous", "Moon phase");
-				if (deltaLongI==270)
+				else if (deltaLongI==270)
 					moonPhase = qc_("Third Quarter", "Moon phase");
-				if (deltaLongI==315)
+				else if (deltaLongI<360)
 					moonPhase = qc_("Waning Crescent", "Moon phase");
-				if (deltaLongI==0 || deltaLongI==360)
-					moonPhase = qc_("New Moon", "Moon phase");
+				else
+					moonPhase = qc_("ERROR IN PHASE STRING PROGRAMMING!", "Moon phase");
 
-				double age = deltaLong*29.530588853/360.;
+				const double age = deltaLong*29.530588853/360.;
 				oss << QString("%1: %2 %3").arg(q_("Moon age"), QString::number(age, 'f', 1), q_("days old"));
 				if (!moonPhase.isEmpty())
 					oss << QString(" (%4)").arg(moonPhase);
 				oss << "<br />";
+
+				// we must repeat the position lookup from above in case we have topocentric corrections.
+				StelUtils::rectToSphe(&ra_equ,&dec_equ, getEquinoxEquatorialPos(core));
+				StelUtils::equToEcl(ra_equ, dec_equ, eclJDE, &lambdaMoon, &betaMoon);
+				StelUtils::rectToSphe(&raSun,&deSun, ssystem->getSun()->getEquinoxEquatorialPos(core));
+				StelUtils::equToEcl(raSun, deSun, eclJDE, &lambdaSun, &betaSun);
+				const double chi=atan2(cos(deSun)*sin(raSun-ra_equ), sin(deSun)*cos(dec_equ)-cos(deSun)*sin(dec_equ)*cos(raSun-ra_equ));
+				oss << QString("%1: %2").arg(q_("PA of bright limb"), StelUtils::radToDecDegStr(StelUtils::fmodpos(chi, M_PI*2.0))) << "<br/>";
```

Comment:
Maybe `Position angle of bright limb` to avoid troubles for newbies?

Reply:
Hmm, I just wanted a shorter label. But OK.